### ISSUE TITLE
Player reports he can't teleport

### ISSUE THREAD

**Issue by MithrandirCraft** [EssentialsX/Essentials#3554]
**Full output of `/ess version`**:

```
> ess version
[17:41:18 INFO]: CONSOLE issued server command: /ess version
[17:41:18 INFO]: Server version: 1.16.1-R0.1-SNAPSHOT git-Paper-127 (MC: 1.16.1)
[17:41:18 INFO]: EssentialsX version: 2.18.0.0
[17:41:18 INFO]: LuckPerms version: 5.1.55
[17:41:18 INFO]: PlaceholderAPI version: 2.10.6
[17:41:18 INFO]: Vault version: 1.7.3-b131
[17:41:18 INFO]: EssentialsXSpawn version: 2.18.0.0
```

**Server log**: Trust me there's absolutely nothing relevant here. The plugin loads fine, and there's no errors when teleport commands are executed, or teleports should occur.

**EssentialsX config**: I don't think this is relevant. But ask for it if so.

### Details

**Description**  
https://youtu.be/Urdhg2IyL-k
This specific user executes any teleport command, but as long as the teleport command has been executed by him, he won't get
teleported.

Tpahere offers from other users towards him will indeed work.
Forcing tping him will also work from an admin.
Other plugin tp's such as WildTP (random teleportation) do work for him, executed from a sign in spawn.

Basically only his own teleport requests from essentialsX don't work.

**Steps to reproduce**  
No idea, this only happens to 1 player. But in essence:
He tries to execute any teleport command, but doesn't get teleported. The countdown works fine, it's just the teleport that doesn't happen.

**Expected behavior**  
User should get teleported after request + cooldown.

**Comment by pop4959** [comment]
Have you tried removing all plugins except for Essentials? This doesn't really sound like a bug that should happen.

**Comment by MithrandirCraft** [comment]
> 
> 
> Have you tried removing all plugins except for Essentials? This doesn't really sound like a bug that should happen.

I'll try that if he still has the issue today.
I know, it looks very random.

**Comment by nullXer0** [comment]
Some of the players on my server experienced the same issue. Restarting the server fixed it.
I enabled debug mode, but it didn't show any errors, it just checked if the player had permissions to use the command, then never attempted to teleport.
I have a 3 second delay before teleporting.

```
>.... [19:35:12 INFO]: 0oxXSinbadXxo0 issued server command: /home
>.... [19:35:12 INFO]: [Essentials] checking if 0oxXSinbadXxo0 has essentials.mail - true
>.... [19:35:12 INFO]: [Essentials] checking if 0oxXSinbadXxo0 has essentials.home - true
>.... [19:35:12 INFO]: [Essentials] checking if 0oxXSinbadXxo0 can afford charge.
>.... [19:35:12 INFO]: [Essentials] calculated command (home) cost for 0oxXSinbadXxo0 as 0
>.... [19:35:12 INFO]: [Essentials] checking if 0oxXSinbadXxo0 can afford charge.
>.... [19:35:12 INFO]: [Essentials] calculated command (home) cost for 0oxXSinbadXxo0 as 0
>.... [19:35:12 INFO]: [Essentials] calculated command (home) cost for 0oxXSinbadXxo0 as 0
>.... [19:35:12 INFO]: [Essentials] checking if 0oxXSinbadXxo0 has essentials.teleport.timer.bypass - false
>.... [19:35:12 INFO]: [Essentials] checking if 0oxXSinbadXxo0 has essentials.teleport.timer.bypass - false
>.... [19:35:12 INFO]: [Essentials] checking if 0oxXSinbadXxo0 has essentials.teleport.timer.move - false
>.... [19:35:12 INFO]: [Essentials] attempting to charge user 0oxXSinbadXxo0
>.... [19:35:12 INFO]: [Essentials] calculated command (home) cost for 0oxXSinbadXxo0 as 0
>.... [19:35:12 INFO]: [Essentials] charge user 0oxXSinbadXxo0 completed
```

```
>.... [19:52:44 INFO]: Server version: 1.16.1-R0.1-SNAPSHOT git-Paper-128 (MC: 1.16.1)
>.... [19:52:44 INFO]: EssentialsX version: 2.18.0.0
>.... [19:52:44 INFO]: LuckPerms version: 5.1.84
>.... [19:52:44 INFO]: Vault version: 1.7.3-b131
>.... [19:52:44 INFO]: ChestShop version: 3.11 (build 213)
>.... [19:52:44 INFO]: EssentialsXProtect version: 2.18.0.0
>.... [19:52:44 INFO]: EssentialsXChat version: 2.18.0.0
>.... [19:52:44 INFO]: Towny version: 0.96.2.0
>.... [19:52:44 INFO]: EssentialsXSpawn version: 2.18.0.0
```

**Comment by MithrandirCraft** [comment]
It happened to a second player, and indeed restarting the server fixes this.
Hence testing by removal of plugins is going to be a bit complicated, since it fixes itself after restarting anyway.

**Comment by pop4959** [comment]
May be similar to #3563. @MithrandirCraft Please provide logs/config though or we will not be able to attempt to replicate this problem.

**Comment by TosieG** [comment]
I can confirm this bug. One of the players reported the same problem to me. Tpa, home, spawn didn't work for him. Server restart fixed it.

**Comment by FrankHeijden** [comment]
I can also confirm this bug on MC 1.16.1 with EssentialsX `2.18.0.1` and `2.18.0.18`. Players reporting they cannot do /tpa, home, and spawn. The teleport message is sent, but the teleport never happens. We also have a delay of 3 seconds enabled for the teleport to commence:
![2020-08-06_11 07 36](https://user-images.githubusercontent.com/22407829/89650107-2b4dde80-d8c2-11ea-9236-9b6bf1ebbbfc.png)

Happens (seems to be) randomly and not to all players.

**Comment by TryToKnow** [comment]
i can confirm this bug, since like 1 year PLEASE FIX IT !!!

**Comment by MithrandirCraft** [comment]
I think this was fixed a few weeks ago after certain dev build? Not sure, since it only happens occasionally and can't be reproduced consistently.

**Edit:** Updated statement unless it wasn't clear enough.

**Comment by JRoy** [comment]
@MithrandirCraft can you please share your config, it is relevant here

**Comment by MithrandirCraft** [comment]
I've not received any complaints on this issue since it was stated as fixed in https://github.com/EssentialsX/Essentials/issues/3563
Not really confirmable that the issue is gone though, since more time should pass, and my server doesn't really have that many players to begin with, in order to be sure.
You still want my config?

**Comment by StartForKiller** [comment]
I can also confirm this bug on MC 1.16.2 with Essentials on 2.18.1.12. Removing the delay on the config seems to solve the issue.

**Comment by Chew** [comment]
Send configs!! If it's broken, we need to see config

**Comment by StartForKiller** [comment]
@Chew 
[config.zip](https://github.com/EssentialsX/Essentials/files/5288827/config.zip)

**Comment by StartForKiller** [comment]
that's the main config, changing the delay to 0 seems to solve the issue

**Comment by Octol1ttle** [comment]
Use a [pastebin](https://hasteb.in)

**Comment by StartForKiller** [comment]
https://hasteb.in/ivafozar.yaml
Done

**Comment by TomatittoChorry** [comment]
cual es la solucion al problema?

**Comment by JRoy** [comment]
Hello! Could you please try the latest dev build to see if it fixes this issue?

**Comment by StartForKiller** [comment]
> Hello! Could you please try the latest dev build to see if it fixes this issue?

I will use it, if it happens to any players i will share the problem again.

**Comment by JRoy** [comment]
Please reopen/comment if this issue is able to be reproduced

**Comment by DotDebian** [comment]
Reopening since I got this issue on the latest build.
We're running 300+ players so I got no way to know what he did.
Enabled debug and here are the logs:
```[22:17:36] [INFO]: [Server Thread] miisplacing issued server command: /home spawner
[22:17:36] [INFO]: [Server Thread] Searching for alternative to: home
[22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.mail - false
[22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.home - true
[22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
[22:17:36] [INFO]: [Server Thread] [Essentials] calculated command (home) cost for miisplacing as 0
[22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
[22:17:36] [INFO]: [Server Thread] [Essentials] calculated command (home) cost for miisplacing as 0
[22:17:36] [INFO]: [Server Thread] [Essentials] calculated command (home) cost for miisplacing as 0
[22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
[22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
[22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.move - false

[22:17:07] [INFO]: [Server Thread] miisplacing issued server command: /spawn
[22:17:07] [INFO]: [Server Thread] Searching for alternative to: spawn
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.mail - false
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.spawn - true
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
[22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
[22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
[22:17:07] [INFO]: [Server Thread] [Essentials] looking up groupname of miisplacing - chevalier
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
[22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
[22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
[22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.move - false
```


I would love to help if you let me debug it more in-depth. Maybe include some more debug in a new build ?
Please note it happens to random player, at a random point and forever until restart.

**Comment by JRoy** [comment]
> Reopening since I got this issue on the latest build.
> We're running 300+ players so I got no way to know what he did.
> Enabled debug and here are the logs:
> 
> ```
> [22:17:36] [INFO]: [Server Thread] Searching for alternative to: home
> [22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.mail - false
> [22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.home - true
> [22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
> [22:17:36] [INFO]: [Server Thread] [Essentials] calculated command (home) cost for miisplacing as 0
> [22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
> [22:17:36] [INFO]: [Server Thread] [Essentials] calculated command (home) cost for miisplacing as 0
> [22:17:36] [INFO]: [Server Thread] [Essentials] calculated command (home) cost for miisplacing as 0
> [22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
> [22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
> [22:17:36] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.move - false
> 
> [22:17:07] [INFO]: [Server Thread] miisplacing issued server command: /spawn
> [22:17:07] [INFO]: [Server Thread] Searching for alternative to: spawn
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.mail - false
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.spawn - true
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
> [22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
> [22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
> [22:17:07] [INFO]: [Server Thread] [Essentials] looking up groupname of miisplacing - chevalier
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing can afford charge.
> [22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
> [22:17:07] [INFO]: [Server Thread] [Essentials] calculated command (spawn) cost for miisplacing as 0
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.bypass - false
> [22:17:07] [INFO]: [Server Thread] [Essentials] checking if miisplacing has essentials.teleport.timer.move - false
> ```
> 
> I would love to help if you let me debug it more in-depth. Maybe include some more debug in a new build ?
> Please note it happens to random player, at a random point and forever until restart.

what is the output of `/ess version`

**Comment by DotDebian** [comment]
![unknown](https://user-images.githubusercontent.com/20359522/118186272-d71def00-b43d-11eb-9cfb-c64f0a130b19.png)

**Comment by JRoy** [comment]
@DotDebian please don't cut off the full message :/

**Comment by DotDebian** [comment]
@JRoy oh yeah I forgot the spigot, it's running PaperSpigot 1.8.8, classic one.

**Comment by JRoy** [comment]
> @JRoy oh yeah I forgot the spigot, it's running PaperSpigot 1.8.8, classic one.

please share a screenshot of the entire message

**Comment by DotDebian** [comment]
![Screenshot_3](https://user-images.githubusercontent.com/20359522/118188354-8a87e300-b440-11eb-8d07-7ef4bd0008ee.jpg)

**Comment by JRoy** [comment]
@DotDebian you're on a version which is over a year old, please update to the latest version found at [essentialsx.net](https://essentialsx.net/downloads.html)

**Comment by DotDebian** [comment]
@JRoy yeah sent the wrong screenshot. Staff told me to update yesterday, which I did. Here's the actual ess version:
![Screenshot_4](https://user-images.githubusercontent.com/20359522/118188949-72649380-b441-11eb-97d9-281965f0237f.jpg)

**Comment by DotDebian** [comment]
Update ? @JRoy

**Comment by JRoy** [comment]
> Update ? @JRoy 

You lied about what version you were on, you should not receive this issue on an up to date version

**Comment by DotDebian** [comment]
> > Update ? @JRoy
> 
> You lied about what version you were on, you should not receive this issue on an up to date version

Getting the issue with the almost up-to-date version of EssentialsX (1 built late)
![image](https://user-images.githubusercontent.com/20359522/118572579-10799600-b781-11eb-84e7-28b279b455d7.png)

Also tried to get smh out of /essentials debug but it just keeps spamming `Constructing new userfile from base player ` so it's unreadable
